crates/matrix-sdk-ui/src/room_list_service/room.rs

```diff
@@ -117,21 +117,22 @@ impl Room {
         self.inner.sliding_sync.unsubscribe_from_room(self.inner.room.room_id().to_owned())
     }
 
-    /// Get the timeline of the room.
-    pub async fn timeline(&self) -> Arc<Timeline> {
+    /// Get the timeline of the room, with an optional timeline event filter.
+    pub async fn timeline(
+        &self,
+        event_filter: Option<Box<TimelineEventFilterFn>>,
+    ) -> Arc<Timeline> {
```

Comment:
I'm not quite happy with this API: if you call this function the first time with `event_filter = None`, then a second time with `event_filter = Some(filter)`, then you might be surprised that the timeline isn't filtered with your filter (because the parameter is used only the first time). Maybe it'd be better to use a pattern with explicit initialization here? so a new function `async fn init_timeline(&self, event_filter: ...)` and this function can remain as `async fn timeline(&self) -> Arc<Timeline>`?

cc @Hywan who might have opinions too

Reply:
Well, I actually asked you about this and you said to use the extra param instead 😅 . Neither of us saw the issue you mentioned above, I guess.

I'm not really sure how to both get a cached timeline and do the filtering in that timeline, should I add some kind of filter -> timeline instance cache? If so, how?

Reply:
Well, this new parameter is… embarrassing. I don't like it here.

Reply:
I'm open to suggestions on how to implement this properly.